### PR TITLE
Shorten paths in VS installation

### PR DESCRIPTION
**Pull request by davidwengier**
Contributes to [AB#1433021](https://devdiv.visualstudio.com/0bdbc590-a062-4c3f-b0f6-9383f67865ee/_workitems/edit/1433021) but the final fix will be in the VS insertion 

Will do a val build and insert it into https://devdiv.visualstudio.com/DevDiv/_git/VS/pullrequest/364093 for testing

### CODE REVIEW

**Comment by davidwengier** [comment]
Val build https://devdiv.visualstudio.com/DefaultCollection/DevDiv/_build/results?buildId=5439629&view=results

**Comment by davidwengier** [comment]
VS build to validate: https://devdiv.visualstudio.com/DevDiv/_git/VS/pullrequest/364093

File: eng/targets/Settings.props [hunk 1]
```diff
@@ -11,7 +11,7 @@
     <VSSDKTargetPlatformRegRootSuffix>RoslynDev</VSSDKTargetPlatformRegRootSuffix>
 
     <CommonExtensionInstallationRoot>CommonExtensions</CommonExtensionInstallationRoot>
-    <LanguageServicesExtensionInstallationFolder>Microsoft\ManagedLanguages\VBCSharp\LanguageServices</LanguageServicesExtensionInstallationFolder>
+    <LanguageServicesExtensionInstallationFolder>Microsoft\VBCSharp\LanguageServices</LanguageServicesExtensionInstallationFolder>
```
Comment (CyrusNajmabadi) [inline eng/targets/Settings.props:14]: i'm curious why we don't just reduce this to `Microsoft\VBCSharp`.  Is the LanguageServices part helpful?

**Review by CyrusNajmabadi — APPROVED** [review]

File: eng/targets/Settings.props [hunk 2]
```diff
@@ -11,7 +11,7 @@
     <VSSDKTargetPlatformRegRootSuffix>RoslynDev</VSSDKTargetPlatformRegRootSuffix>
 
     <CommonExtensionInstallationRoot>CommonExtensions</CommonExtensionInstallationRoot>
-    <LanguageServicesExtensionInstallationFolder>Microsoft\ManagedLanguages\VBCSharp\LanguageServices</LanguageServicesExtensionInstallationFolder>
```
Comment (CyrusNajmabadi) [inline eng/targets/Settings.props:14]: asdad

File: eng/targets/Settings.props (hunk 1 above)
Comment (davidwengier) [reply eng/targets/Settings.props:14]: Both `LanguageServices` and `ExpressionEvaluator` are extensions that have been extracted, so they have a few overlapping files, like `manifest.json` etc.

I did propose on an email thread shortening them to `LS` and `EE`, or we could do `VBCSLanguageServices` and `VBCSExpressionEvaluator` directly under the `Microsoft` folder. To be perfectly frank the "Microsoft" bit seems a little redundant even :)

**Comment by genlu** [comment]
@davidwengier Could you please validate our .Net 6 OOP still work with this change? Especially our dependent services (Test discovery, razor, etc.), which used to use relative path to Roslyn for assembly probing. It was removed a while ago, so our OOP services shouldn't be affected. But just in case :)

**Review by jasonmalinowski — APPROVED** [review]
I can't wait to see what components have improperly taken dependencies on this!

**Review by JoeRobich — APPROVED** [review]

**Comment by davidwengier** [comment]
Well looks like Razor is broken, though I don't know why. Using the .NET 6 OOP process doesn't seem to make it better or worse.

**Comment by RikkiGibson** [comment]
This gets me wondering if win11 is changing any defaults around long paths.

**Comment by davidwengier** [comment]
Update: The razor issue is https://dev.azure.com/devdiv/DevDiv/_workitems/edit/1433840 so going to rebase the VS branch and try again.

**Comment by davidwengier** [comment]
Razor works! 🎉

As does LUT, test discovery, with .NET 6 host and non. Seems normal to me :)

**Comment by genlu** [comment]
@davidwengier Thanks for the validation!

**Review by genlu — APPROVED** [review]

**Comment by davidwengier** [comment]
This has passed manual testing :D

**Comment by genlu** [comment]
FYI @y87feng, who's fixing a similar issue and could make the same change for LUT and SBD.

**Comment by davidwengier** [comment]
> FYI @y87feng, who's fixing a similar issue and could make the same change for LUT and SBD.

Thanks @genlu we actually validated this change with LUT and SBD in the same VS branch :)

**Comment by davidwengier** [comment]
Change to release/dev17.1. @jinujoseph for M2

**Pull request merged**